phper/src/builder/argument.rs

```diff
@@ -0,0 +1,88 @@
+use crate::builder::data_type::AtomicType;
+use crate::builder::describe::Describe;
+use crate::builder::value::ConstantValue;
+
+pub enum ArgumentFlag {
+    IsVaradic,
+    HasDefault(ConstantValue),
+}
+
+pub struct Argument {
+    pub name: String,
+    pub data_type: AtomicType,
+    pub flag: Option<ArgumentFlag>,
```

Comment:
`varadic` and `default` can't go together, hence why they are behind a flag, while it is mentioned in the issue that those objects should be treated as "dirty", we should try to make sure that users make as little mistakes as possible, one of which is defining a varadic argument with a default.

note: technically, varadic arguments have a default, and that default is an empty list ( i.e `ConstantValue::List(vec![])` ), which we should remember when parsing.